### PR TITLE
Add long pointy sticks and things they can be used for

### PR DESCRIPTION
**Pull request by Alterforce**
#### Summary

Content "Long Pointy Sticks and things to use them"

#### Purpose of change
This change allows players to craft a simple weapon has reach without the use of fire.
Long pointy sticks can also be used for constructions and recipes that call for wooden spears or pointy sticks, and long pointy sticks can be used to craft wooden spears.

#### Describe the solution
Allow survivors to turn long sticks into long pointy sticks, like how sticks can be turned into pointy sticks.

#### Describe alternatives you've considered
Not adding the long pointy stick.

#### Testing
I have tested the long pointy stick in game by crafting it and using it in crafting recipes and constructions along with fighting zombies with it, no crashes occurred.

#### Additional context
I have modeled the long pointy stick's stats (weight, length, etc.) on a halfway point between the long stick and the wooden spear and the damage is modeled on the pointy stick, with the additional bash damage due to having more mass than a pointy stick.

### CODE REVIEW

**Comment by Maleclypse** [comment]
To lint you json copy each entire file into this website https://dev.narc.ro/cataclysm/format.html  hit lint and then copy the result back into your json editor.  It doesn't say anything if it works and if it says syntax error you can find the error using jsonlint.com.   It's important to realize we use a different json dialect than jsonlint.com so you'd need to lint anything from jsonlint.com in the https://dev.narc.ro/cataclysm/format.html.

**Review by Alterforce — COMMENTED** [review]
I have linted construction.json and piercing.json.

**Comment by Alterforce** [comment]
Build and test process completed with exit code 3, what should I do?

File: data/json/items/melee/spears_and_polearms.json [hunk 1]
```diff
@@ -38,6 +38,26 @@
...
+    "to_hit": -1,
+    "price_postapoc": 10,
+    "color": "brown",
+    "looks_like": "spear_wood",
+    "symbol": "/",
+    "material": [ "wood" ],
+    "techniques": [ "WBLOCK_1" ],
+    "volume": "2500 ml",
+    "longest_side": "205 cm",
+    "bashing": 9,
```
Comment (kevingranade) [inline data/json/items/melee/spears_and_polearms.json:56]:
```suggestion
    "bashing": 5,
```

**Comment by kevingranade** [comment]
Construction recipes sometimes trigger a piece of obscure infrastructure for handling faction blueprints, I'll look into it.

**Comment by kevingranade** [comment]
If you rebase or just re-push after #51521 is merged it should work.

**Pull request merged**